clang/tools/dpct/codepin/codepin-report.py

```diff
@@ -340,6 +369,144 @@ def get_memory_used(cp_list):
     return (cp_id, max_mem)
 
 
+def generate_data_flow_graph(
```

Comment:
Please add comment to this function,  highlight the main structure of the Alrogithm, in case end user may custom it or update it.